doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,154 @@
+.. _compil-steps:
+
+========================
+Command level processing
+========================
+
+When a command, for instance :g:`Check 1 + 2 * 2.` is fed to Rocq, it
+goes through several *successive* processing steps.
+
+Lexing
+~~~~~~
+
+The first step, :gdef:`lexing` splits the input into a sequence of
+tokens, for instance the string ``"Check 1 + 2 * 2."`` is split into
+the sequence of seven tokens `'Check' '1' '+' '2' '*' '2' '.'`. A set
+of :ref:`basic tokens <lexical-conventions>` are predefined and can be
+extended, in particular by
+:ref:`reserving notations <ReservingNotations>`.
+
+Parsing
+~~~~~~~
+
+During :gdef:`parsing` the sequence of tokens is interpreted as a
+tree, for instance here:
+
+.. code-block:: text
+   :name: after-parsing
+
+         Check
+           |
+           +
+          / \
+         1   *
+            / \
+           2   2
+
+The parsed grammar can be modified, for instance by
+:ref:`reserving notations <ReservingNotations>`.
+
+Synterp
+~~~~~~~
+
+In addition to the above lexing and parsing, the :gdef:`synterp` phase
+does just enough job to be able to parse the remaining commands.
```

Comment:
```suggestion
does just enough to be able to parse the remaining commands.
```
I would like this paragraph to be more precise. I don't really understand what is happening during `synterp` by reading it. I can't really help, since I really don't know what it does.

Reply:
I've added a short sentence mentioning plugins, Require and Import.
Also added a small note mentioning that this synterp/interp distinction is important for developers but shouldn't matter too much for users.